spec/models/bundle_context_spec.rb

```diff
@@ -61,10 +64,97 @@
       end
     end
 
+    context "bundle_dir path does not exist" do
+      let(:incorrect_bc) do
+        BundleContext.new(
+          project_name: "SmokeTest",
+          content_structure: 1,
+          bundle_dir: "does/not/exist",
+          staging_style_symlink: false,
+          content_metadata_creation: 1,
+          user: user
+        )
+      end
+
+      it "object does not pass validation" do
+        expect(incorrect_bc).not_to be_valid
+      end
+
+    end
+
     it { is_expected.to validate_presence_of(:project_name) }
     it { is_expected.to validate_presence_of(:content_structure) }
     it { is_expected.to validate_presence_of(:bundle_dir) }
     it { is_expected.to validate_presence_of(:content_metadata_creation) }
     it { is_expected.to belong_to(:user) }
   end
+
+  describe "#staging_dir" do
+    it 'is hardcoded to the correct path' do
+      expect(described_class.new.staging_dir).to eq '/dor/assembly'
+    end
+  end
+
+  describe "#content_tag_override?" do
+    it "is set to true" do
+      expect(described_class.new.content_tag_override?).to be true
+    end
+  end
+
+  describe "#smpl_manifest" do
+    it "returns the file name" do
+      expect(described_class.new.smpl_manifest).to eq 'smpl_manifest.csv'
+    end
+  end
+
+  describe "#manifest" do
+    it "returns the file name" do
+      expect(described_class.new.manifest).to eq 'manifest.csv'
+    end
+  end
+
+  describe "#path_in_bundle" do
+    it "creates a relative path" do
+      expect(bc.path_in_bundle("manifest.csv")).to eq "spec/test_data/bundle_input_g/manifest.csv"
+    end
+  end
+
+  describe "#progress_log_file" do
+    skip("Need to figure out where to set this path via planning meeting 9/10/18")
+  end
+
+  describe '#import_csv' do
+    let(:manifest) do
+      described_class.import_csv("#{Rails.root}/spec/test_data/bundle_input_a/manifest.csv")
+    end
+
+    it "loads a CSV as a hash with indifferent access" do
+      expect(manifest).to be_an(Array)
+      expect(manifest.size).to eq(3)
+      headers = %w{format sourceid filename label year inst_notes prod_notes has_more_metadata description}
+      expect(manifest).to all(be_an(ActiveSupport::HashWithIndifferentAccess)) # accessible w/ string and symbols
+      expect(manifest).to all(include(*headers))
+      expect(manifest[0][:description]).to be_nil
+      expect(manifest[1][:description]).to eq('')
+      expect(manifest[2][:description]).to eq('yo, this is a description')
+    end
+  end
+
+  describe "manifest_rows" do
```

Comment:
i'd be inclined to add a couple tests here:
1. something integration-y that tests the expected content of `manifest_rows`
2. a test for the memoization

Reply:
added both!